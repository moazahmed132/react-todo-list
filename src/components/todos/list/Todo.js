import React, { Component, Fragment } from 'react'

class Todo extends Component {
  state = {
    isEdit: false

  }

  renderTodo = (props) => {
    return (
      <li className="todo-list">
        <span>{this.props.details.name}</span>
        <i className="fas fa-trash-alt " onClick={() => { this.props.deleteTodo(this.props.index) }}></i>
        <i className="fas fa-pen-square " onClick={() => { this.toggleState() }}  ></i>
        {/* <input type="checkbox" onClick={() => { this.props.doneTodo(this.props.index) }} ></input>*/}
        {<i className="far fa-circle" onClick={() => { this.props.doneTodo(this.props.index) }}></i>}

      </li>

    )
  }

  updateTodoItem = (e) => {
    e.preventDefault();
    this.props.editTodo(this.props.index, this.input.value)
    if (this.input.value.trim() !== "") {
      this.toggleState()
    }
  }

  renderUpdateForm = () => {
    return (
      <form onSubmit={this.updateTodoItem}>
        <input className="form-input"
          type="text" ref={(v) => { this.input = v }} defaultValue={this.props.details.name} />
        <button className="myButton">Update</button>
      </form>
    )
  }

  toggleState = () => {
    let { isEdit } = this.state;
    this.setState({
      isEdit: !isEdit
    })
  }

  render() {
    let { isEdit } = this.state
    return (
      <Fragment >
        {isEdit ? this.renderUpdateForm() : this.renderTodo()}
      </Fragment>
    );
  }
}

export default Todo