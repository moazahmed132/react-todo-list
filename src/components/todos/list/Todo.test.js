import React from 'react';
import Todo from './Todo';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


describe('Todo Component', () => {
  let wrapper;
  let props;

  Enzyme.configure({
    adapter: new Adapter()
  });

  beforeEach(() => {
    props = {
      onChange: jest.fn()
    }

    wrapper = shallow(
      <Todo />)
  })
  it('Should render the right way', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the right way', () => {
    expect(wrapper.find('li')).to.have.lengthOf(1);
  });

  it('Should render the right way', () => {
    expect(wrapper.find('i')).to.have.lengthOf(3);
  });

})

