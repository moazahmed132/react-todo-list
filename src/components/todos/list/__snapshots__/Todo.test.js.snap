// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component Should render the right way 1`] = `undefined`;
